pkg/operation/seed/seed.go

```diff
@@ -739,6 +739,7 @@ func RunReconcileSeedFlow(
 	if err != nil {
 		return err
 	}
+	anySNI = anySNI || gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI)
```

Comment:
Probably it is better to rename this to
```suggestion
	sniEnabledOrInUse = anySNI || gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI)
```